### DISCUSSION TITLE
Feat: add indexes to columns

### DISCUSSION THREAD

**Discussion by kiwicopple** [supabase/supabase#13772] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.

It would be great to be able to add an index to a column

## Describe the solution you'd like

Perhaps a right-click > Add index > Sidebar / dialog?

## Additional context

We need to consider a few more items:

- A user should be able to add a multi-index column somehow. [docs](https://www.postgresql.org/docs/current/indexes-multicolumn.html)
- A user should be able to choose the type of index: [docs](https://www.postgresql.org/docs/9.1/indexes-types.html)
- Eventually we need to support [all of the functionality](https://www.postgresql.org/docs/current/indexes.html) of Postgres indexes, but I think the lowest hanging fruit is just a single-column index where they can choose between `B-tree, Hash, GiST, SP-GiST, GIN and BRIN`

**Comment by geoguide** [comment]
100% agree and it'd be even better if it allowed for adding a composite index too since I do that all the time and it's hard to view them.

Update: This is avilable in the database section, but i expected it to be available when editing a column. The features are there but the UX could use more clarity.

**Reply by Ibby-debug** [reply]
it took me around 20 mins to figure out where the hell it is!
